contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -185,4 +197,63 @@ contract OrderValidationUtils is
         transferableAssetAmount = LibSafeMath.min256(balance, allowance);
         return transferableAssetAmount;
     }
+
+    /// @dev This function handles the edge cases around taker validation. This function
+    ///      currently attempts to find duplicate ERC721 token's in the taker
+    ///      multiAssetData.
+    /// @param assetData The asset data that should be validated.
+    /// @return Whether or not the order should be considered valid.
+    function _isAssetDataValid(bytes memory assetData)
+        internal
+        pure
+        returns (bool)
+    {
+        // Asset data must be composed of an asset proxy Id and a bytes segment with
+        // a length divisible by 32.
+        if (assetData.length % 32 != 4) {
+            return false;
+        }
+
+        // Only process the taker asset data if it is multiAssetData.
+        bytes4 assetProxyId = assetData.readBytes4(0);
+        if (assetProxyId != IAssetData(address(0)).MultiAsset.selector) {
+            return true;
+        }
+
+        // Get array of values and array of assetDatas
+        (, uint256[] memory assetAmounts, bytes[] memory nestedAssetData) = decodeMultiAssetData(assetData);
+
+        uint256 length = nestedAssetData.length;
+        for (uint256 i = 0; i != length; i++) {
+            // TODO(jalextowle): Implement similar validation for non-fungible ERC1155 asset data.
```

Comment:
Should we handle the other cases now and return `false` or is it convention to let it pass through?

Reply:
When you say "handle the other cases now", do you mean in this PR, or at that part of the code? We can't just return false, because then everything will be marked as invalid.

Reply:
Ah, okay. Yeah I meant wrt validation. So basically if we don't recognize something then it is automatically valid?

Reply:
Yep. Otherwise, we'd have to recognize asset data for it to be valid, which I don't think makes sense. As of right now, we are never going to be able to recognize all asset data. This is important when you consider that people can't post orders to 0x-mesh unless they are valid. I don't think that we want to spend a ton of time *right now* writing more bespoke validation logic, so I think the best course of action is probably to implement a permissive version.

I'm very open to thoughts and criticisms of this argument though.